creation/web_base/cvmfs_helper_funcs.sh

```diff
@@ -30,476 +30,522 @@
+    loginfo "Kernel major revision: $GWMS_OS_KRNL_MAJOR_REV"
+    loginfo "Kernel minor revision: $GWMS_OS_KRNL_MINOR_REV"
+    loginfo "Kernel patch number: $GWMS_OS_KRNL_PATCH_NUM"
+
+    loginfo "CVMFS locally installed: $(check_exit_status $GWMS_IS_CVMFS_LOCAL_MNT)"
+    loginfo "Unprivileged user namespaces supported: $(check_exit_status $GWMS_IS_UNPRIV_USERNS_SUPPORTED)"
+    loginfo "Unprivileged user namespaces enabled: $(check_exit_status $GWMS_IS_UNPRIV_USERNS_ENABLED)"
+    loginfo "FUSE installed: $(check_exit_status $GWMS_IS_FUSE_INSTALLED)"
+    loginfo "fusermount available: $(check_exit_status $GWMS_IS_FUSERMOUNT)"
+    loginfo "Is the user in 'fuse' group: $(check_exit_status $GWMS_IS_USR_IN_FUSE_GRP)"
+    loginfo "..."
 }
 
 
 mount_cvmfs_repos () {
-        # DESCRIPTION: This function mounts all the required and additional
-        # CVMFS repositories that would be needed for user jobs.
-        #
-        # INPUT(S): 1. CVMFS configuration repository (string); 2. Additional CVMFS
-        # repositories (colon-delimited string)
-        # RETURN(S): Mounts the defined repositories on the worker node filesystem
-
-	$glidein_cvmfsexec_dir/$dist_file $1 -- echo "setting up mount utilities..." &> /dev/null
-	if [[ $(df -h|grep /cvmfs|wc -l) -eq 1 ]]; then
-		loginfo "CVMFS config repo already mounted!"
-		continue
-	else
-		# mounting the configuration repo (pre-requisite)
-		loginfo "Mounting CVMFS config repo now..."
-		$glidein_cvmfsexec_dir/.cvmfsexec/mountrepo $1
-	fi
-
-	# using an array to unpack the names of additional CVMFS repositories
-	# from the colon-delimited string
-	declare -a cvmfs_repos
-	repos=($(echo $2 | tr ":" "\n"))
-	#echo ${repos[@]}
-
-	loginfo "Mounting additional CVMFS repositories..."
-	# mount every repository that was previously unpacked
-	for repo in "${repos[@]}"
-	do
-		$glidein_cvmfsexec_dir/.cvmfsexec/mountrepo $repo
-	done
-
-	# see if all the repositories got mounted
-	num_repos_mntd=`df -h | grep /cvmfs | wc -l`
-	total_num_repos=$(( ${#repos[@]} + 1 ))
-	if [ "$num_repos_mntd" -eq "$total_num_repos" ]; then
-		loginfo "All CVMFS repositories mounted successfully on the worker node"
-		true
-	else
-		logwarn "One or more CVMFS repositories might not be mounted on the worker node"
-		false
-	fi
-
-	GWMS_IS_CVMFS=$?
+    # DESCRIPTION: This function mounts all the required and additional
+    # CVMFS repositories that would be needed for user jobs.
+    #
+    # INPUT(S): 1. CVMFS configuration repository (string); 2. Additional CVMFS
+    # repositories (colon-delimited string)
+    # RETURN(S): Mounts the defined repositories on the worker node filesystem
+    local cvmfsexec_mode=$1
+    local config_repository=$2
+    local additional_repos=$3
+
+    if [[ $cvmfsexec_mode -eq 1 ]]; then
+        "$glidein_cvmfsexec_dir/$dist_file" "$1" -- echo "setting up mount utilities..." &> /dev/null
+    fi
+    if [[ $(df -h|grep /cvmfs|wc -l) -eq 1 ]]; then
```

Comment:
This is only checking whether the config repo was mounted as part of what happened in the previous `if` block. At this point in time, only one repo would be mounted which is the config repo, therefore `-eq` is used (and not `ge`).

If there are more CVMFS repos already mounted, then that would basically mean that CVMFS is already available locally which means this part of the code will not be executed.

Reply:
add a comment about the tehcnical detail that at this point in execution flow, it would have already been determined that cvmfs is not locally available, so zero repositories are mounted and just config repo is being mounted here.

Reply:
updated as suggested above.